### DISCUSSION TITLE
How can I decrypt a JWT payload if the token is expired?

### DISCUSSION THREAD

**Discussion by LucaIgnatescu** [panva/jose#680] [Q&A]
The workaround I found was not specifying an exp claim, and calling decryptJWT twice, first with a maxTokenAge, and then again without one in case of an ERR_JWT_EXPIRED error. Is there any better way to do this, ideally with a jose function directly?

**Comment by panva** [comment]
I've provided means to access the parsed JWT payload attached to all `JWTClaimValidationFailed` and `JWTExpired` error instances in https://github.com/panva/jose/releases/tag/v5.4.0
